### PR TITLE
Update to the latest Linebender CI standard.

### PR DESCRIPTION
**Pull request by xStrom**
* Enable caching (improves compile step performance by ~50%)
* Add typos check
* Minor tweaks to match the latest standard

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ env:
   # This should be the same version of Rust which the above stable was branched from.
   # To get the correct date, use `cargo --version`. E.g for 1.77, this prints
   # cargo 1.77.0 (3fe68eabf 2024-02-29)
-  # So the date used when we depended on 1.77 was 2024-02-29
-  RUST_DOCS_COMPILE_VER: "nightly-2024-02-29"
+  # So the date used when we depended on 1.77 was 2024-02-29.
+  RUST_DOCS_COMPILE_VER: "nightly-2024-06-03"
```
Comment (DJMcNab) [inline .github/workflows/ci.yml:15]: We probably should bump the stable version as well

**Review by DJMcNab — APPROVED** [review]
Thanks!

**Comment by xStrom** [comment]
Caching was previously disabled here, due to performance concenrs:
> Our dependency tree is small enough (only libc) that caching is likely to be worse

Motivated by wanting fewer differences between repos I wanted to see if this remains true today. The performance numbers I got are the following:


| scenario | restore | work | store | total | diff | warm total vs cold work diff |
| --- | --- | --- | --- | --- | --- | --- |
| cargo check (msrv) cold cache | 0s | 7s | 2s | 9s | |	|
| cargo check (msrv) warm cache | 1s | 2s | 0s | 3s | -67% | -57% |
| cargo clippy (armv7) cold cache | 0s | 8s | 2s | 10s | | |
| cargo clippy (armv7) warm cache | 0s | 5s | 0s | 5s | -50% | -38% |
| cargo clippy (aarch64) cold cache | 0s | 8s | 1s | 9s | | |
| cargo clippy (aarch64) warm cache | 1s | 4s | 0s | 5s | -44% | -38% |
| cargo clippy (x86_64) cold cache | 0s | 10s | 1s | 11s | | |
| cargo clippy (x86_64) warm cache | 1s | 4s | 0s | 5s | -55% | -50% |
| cargo doc cold cache | 1s | 5s | 1s | 7s | | |
| cargo doc warm cache | 3s | 0s | 0s | 3s | -57% | -40% |
| cargo test --doc cold cache | 0s | 6s | 2s | 8s | | |
| cargo test --doc warm cache | 1s | 1s | 0s | 2s | -75% | -67% |

Turns out that the slowness of the Rust compiler should not be underestimated and even with such a small dependency tree there are significant time savings. Yay! 🎉

File: .github/workflows/ci.yml (hunk 1 above)
Comment (xStrom) [reply .github/workflows/ci.yml:15]: Yeah, I'll eventually be doing a pass on all the repos to make sure the versions are bumped. I'm omitting that in this script sync pass to reduce PR size.

**Review by DJMcNab — APPROVED** [review]

**Pull request merged**